Blueprints/BlueprintDefinitions/Msbuild-NETCore_2_1/EmptyFunctionDepInjection/template/test/BlueprintBaseName.1.Tests/FunctionTest.cs

```diff
@@ -0,0 +1,25 @@
+using Amazon.Lambda.Core;
+using Amazon.Lambda.TestUtilities;
+using Microsoft.Extensions.Configuration;
+using Moq;
+using Xunit;
+
+using BlueprintBaseName._1;
+
+namespace BlueprintBaseName._1.Tests
+{
+    public class FunctionTest
+    {
+        [Fact]
+        public void TestConfigFunction()
+        {
+            var expected = "val1";
+            var mockConfig = new Mock<IConfiguration>();
+            mockConfig.Setup(p => p[It.IsAny<string>()]).Returns(expected);
+
+            var function = new Function(mockConfig.Object);
+            var result = function.FunctionHandler("env1", new TestLambdaContext());
+            Assert.Equal(expected, result);
+        }
```

Comment:
You could get away with not using a mock here with something like this:
```csharp
var expected = "val1";
var configValues = new Dictionary<string,string>{ ["env1"] = "val1" };
var testConfig = new ConfigurationBuilder().AddInMemoryCollection(configValues).Build();

var function = new Function(testConfig);
var result = function.FunctionHandler("env1", new TestLambdaContext());
Assert.Equal(expected, result);
```
That way you can drop moq and the template becomes a bit more non-prescriptive.

Reply:
@slang25 Good idea. I'll eliminate the Moq bits.